### ISSUE TITLE
Supporting authentication through a managed identity instead of using a normal app

### ISSUE THREAD

**Issue by Epoxylim** [cudeso/misp2sentinel#3]
Would this be something that could be supported going forward?

**Comment by cudeso** [comment]
Yes, it's certainly something to consider.
I do not have a lot of experience with authenticating via a managed identity but I'll put it on the list to investigate and for future implementations.

**Comment by lnfernux** [comment]
I can probably do this for the azure function app already, it's pretty simple to integrate a authentication chain there. I'll do a PoC and push it as a PR once the upload indicators branch is merged. 

Should be the same lines of code for a vm running in Azure as it is for the Azure Function :)

**Comment by Kaloszer** [comment]
Hey @Infernux, were you able to do that yet? I actually really really want to do it that way :D

**Comment by lnfernux** [comment]
I have a PoC for this locally, but I haven't made it production friendly yet. Will try to get some time to do that the following weeks, I'll update here once I'm able.

**Comment by jusso-dev** [comment]
I noticed that the README mentions Azure Key Vault (only works on Azure VM) - "https://github.com/cudeso/misp2sentinel/tree/main?tab=readme-ov-file#azure-key-vault-integration-only-works-on-azure-vms"

But I'm not sure why this is called out? If the Azure Function has SystemIdentity/Managed System Identity (MSI) turned on, there is no reason it won't work with Azure Key Vault, you just need to configured the provisioned MSI with relevant Azure Key Vault RBAC or Access Policies.

**Comment by jusso-dev** [comment]
Clarified in #86 
@lnfernux

**Comment by cudeso** [comment]
Duplicate of #134
